src/api/db/models/Task.js

```diff
@@ -0,0 +1,101 @@
+import { NotFoundError, ForbiddenError, AuthenticationError } from '../../errors.js';
+import SQS from '@aws-sdk/client-sqs';
+import MongoPaging from 'mongo-cursor-pagination';
+import Task from '../schemas/Task.js';
+import { hasRole } from './utils.js';
+import {
+  READ_TASKS_ROLES
+} from '../../auth/roles.js';
+
+/**
+ * Tasks manage the state of async events (except for batch uploads) on the platform
+ * @class
+ */
+export class TaskModel {
+  /**
+   * Query Tasks by Filter, returning a paged list
+   *
+   * @param {Object} input
+   * @param {String} input.limit
+   * @param {String} input.paginatedField
+   * @param {String} input.sortAscending
+   * @param {String} input.next
+   * @param {String} input.previous
+   * @param {Object} context
+   */
+  static async queryByFilter(input, context) {
+    return await MongoPaging.aggregate(Task.collection, {
+      aggregation: [
+        { '$match': { 'projectId': context.user['curr_project'] } },
+        { '$match': { 'user': context.user.sub } }
```

Comment:
Oh yeah looks like here we are using `user.sub` but in the Task creation we're using `user.username`?